design/baremetal-operator/inspector-deprecation.md

```diff
@@ -0,0 +1,173 @@
+
+## Motivation
+
+The [spec][inspector-merge] covers the reasons in detail, and many of the
+arguments apply to Metal3 as well, for instance:
+
+- Lower maintenance
+- Less configuration code in [ironic-image][ironic-image]
+- Removing the potential point of failure between the two services
+- Probably most importantly, paving a way for easier active-active HA setup
+
+To elaborate on the last point, while Ironic has been designed with high
+availability and load splitting in mind, Inspector was single-process with
+some HA capabilities added as an afterthought.
+
+[ironic-image]: https://github.com/metal3-io/ironic-image/
+
+### Goals
+
+- No separate ironic-inspector container is started.
+- All existing functionality is preserved.
+
+### Non-Goals
+
+- Upgrade process for clouds that don't use [ironic-image][ironic-image] falls
+  under the upstream Ironic project responsibilities.
+
+## Proposal
+
+BMO will be updated to not assume the presence of ironic-inspector. The
+[ironic-image][ironic-image] will be updated to configure Ironic for inspection
+without ironic-inspector.
+
+## Design Details
+
+### Implementation Details/Notes/Constraints
+
+BMO scope:
+
+- Use the new [inventory API][inventory API] instead of accessing
+  ironic-inspector. This API functions both with ironic-inspector and without
+  it, delegating to ironic-inspector if necessary.
+
+- Use only fields on the `Node` object to determine the inspection status
+  and error message.
+
+- Stop creating the ironic-inspector client and validating the service.
+
+- Adjust the hardware data code to support both inspection data formats.
+  They are different between ironic-inspector and the new inspection, but
+  they still provide the same information.
+
+[inventory API]: https://docs.openstack.org/api-ref/baremetal/#show-node-inventory
+
+ironic-image scope:
+
+- Enable the new inspection implementation, make sure the configuration
+  stays largely the same. Some hooks are called differently, we may need
+  to account for that.
+
+- Stop configuring Ironic to talk to ironic-inspector.
+
+- Drop the ironic-inspector scripts, including WSGI support.
+
+### Risks and Mitigations
+
+- Minor configuration differences may affect operators that have non-default
+  configuration. We will need to document these.
+
+- Operators not using ironic-image will need to use upstream migration
+  procedures. They have to be in place by the time we pull the trigger.
+
+- The new inspection work will be finished in the upcoming (as of writing this
+  proposal) Ironic release 2024.1 "Caracal". To accommodate operators wishing
+  to use stable releases, we will either need to keep ironic-inspector support
+  in place until April 2024 or to provide long-term support branches of both
+  BMO and ironic-image with ironic-inspector support. The latter is safer
+  from the standpoint of not breaking existing operators.
```

Comment:
just want to point out that we need to understand what long-term support means here since the current plan for ironic-image branches is to have 1-2 stable branches based on the bugfix ironic branches, that are supported for 6 months
if we go over the 6 months, then we need to redefine entirely the ironic-image branching strategy, for example basing LTS on stable openstack branches and use bugfix branches as intermediate steps

Reply:
Actually... We probably need LTS ironic-image branches to cover the Steve's case, only BMO's. You're not using ironic-image, right @hardys?

Reply:
@dtantsur sorry for the slow response - currently we're consuming the BMO/CAPM3 releases and using downstream Ironic packages based on OpenStack LTS stable branches (which I'm assuming is a fairly common pattern for downstream consumers of Metal<sup>3</sup>?)